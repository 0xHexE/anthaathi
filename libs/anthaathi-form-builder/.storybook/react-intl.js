const locales = ['en-US'];

const messages = locales.reduce(
  (acc, lang) => ({
    ...acc,
    [lang]: import(`../locales/compiled-locales/${lang}.json`), // whatever the relative path to your messages json is
  }),
  {},
);

const formats = {};

export const reactIntl = {
  defaultLocale: 'en-US',
  locales,
  messages,
  formats,
};
