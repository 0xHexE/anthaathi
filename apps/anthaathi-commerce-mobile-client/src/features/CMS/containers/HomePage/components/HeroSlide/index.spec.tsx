import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import HeroSlide from './index';

describe('HeroSlide', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <HeroSlide
            backgroundImageSrc="https://cdn.shopify.com/s/files/1/0648/1303/9842/files/fresh-squeezed-orange-juice_300x.jpg?v=1653584430"
            title="test"
            subTitle="test"
            buttonTitle="View All"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('heroSlide')).toBeTruthy();
  });
});
