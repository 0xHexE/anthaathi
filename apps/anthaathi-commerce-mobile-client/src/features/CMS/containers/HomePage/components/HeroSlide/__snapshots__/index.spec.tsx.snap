// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSlide should render the item 1`] = `
<View
  style={
    {
      "marginVertical": 10,
    }
  }
  testID="heroSlide"
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "height": 240,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://cdn.shopify.com/s/files/1/0648/1303/9842/files/fresh-squeezed-orange-juice_300x.jpg?v=1653584430",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": 240,
            "width": "100%",
          },
          undefined,
        ]
      }
      testID="backgroundImage"
    />
    <View
      style={
        {
          "backgroundColor": "rgba(0,0,0,0.1)",
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "flex-start",
            "display": "flex",
            "flexGrow": 1,
            "justifyContent": "flex-end",
            "marginHorizontal": 20,
            "marginVertical": 30,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "color": "#f6f6f6",
                "fontSize": 24,
                "fontWeight": "bold",
                "marginVertical": 5,
                "paddingTop": 10,
              },
            ]
          }
          variant="titleSmall"
        >
          test
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "color": "#f6f6f6",
                "fontSize": 16,
                "fontWeight": "400",
                "marginVertical": 10,
              },
            ]
          }
          variant="titleSmall"
        >
          test
        </Text>
        <View
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              false,
              undefined,
            ]
          }
          testID="heroSlideOnPress"
        >
          <View
            style={
              {
                "backgroundColor": "#f6f6f6",
                "borderRadius": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": undefined,
                    "fontWeight": "700",
                    "paddingBottom": 8,
                    "paddingLeft": 18,
                    "paddingRight": 18,
                    "paddingTop": 8,
                  },
                ]
              }
              testID="buttonTitle"
              variant="labelSmall"
            >
              View All
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
